autoPyTorch/datasets/time_series_dataset.py

```diff
@@ -1,174 +1,1267 @@
+                                         compute_mase_coefficient_value=False,
+                                         time_features=cached_time_features,
+                                         is_test_set=True)
+
+            return val_set
+
+    def get_test_target(self, test_idx: int) -> np.ndarray:
+        if self.is_test_set:
+            raise ValueError("get_test_target is not supported for test sequences!")
+        if test_idx < 0:
+            test_idx = self.__len__() + test_idx
+        Y_future = self.Y[test_idx + 1: test_idx + self.n_prediction_steps + 1]
+        return Y_future
+
+    def update_attribute(self, **kwargs: Any) -> None:
+        for key, value in kwargs.items():
+            if not hasattr(self, key):
+                raise ValueError('Trying to update invalid attribute for TimeSeriesSequence!')
+            setattr(self, key, value)
+
+
+class TimeSeriesForecastingDataset(BaseDataset, ConcatDataset):
+    datasets: List[TimeSeriesSequence]
+    cumulative_sizes: List[int]
+
+    def __init__(self,
+                 X: Optional[Union[np.ndarray, List[Union[pd.DataFrame, np.ndarray]]]],
+                 Y: Union[np.ndarray, List[Union[pd.DataFrame, np.ndarray]]],
+                 X_test: Optional[Union[np.ndarray, List[Union[pd.DataFrame, np.ndarray]]]] = None,
+                 Y_test: Optional[Union[np.ndarray, List[Union[pd.DataFrame, np.ndarray]]]] = None,
+                 start_times: Optional[List[pd.DatetimeIndex]] = None,
+                 series_idx: Optional[Union[List[Union[str, int]], str, int]] = None,
+                 known_future_features: Optional[Union[Tuple[Union[str, int]], Tuple[()]]] = None,
+                 time_feature_transform: Optional[List[TimeFeature]] = None,
+                 freq: Optional[Union[str, int, List[int]]] = None,
+                 resampling_strategy: Optional[
+                     Union[CrossValTypes, HoldoutValTypes, NoResamplingStrategyTypes]
+                 ] = HoldoutValTypes.time_series_hold_out_validation,
                  resampling_strategy_args: Optional[Dict[str, Any]] = None,
-                 shuffle: Optional[bool] = False,
+                 shuffle: Optional[bool] = True,
                  seed: Optional[int] = 42,
                  train_transforms: Optional[torchvision.transforms.Compose] = None,
                  val_transforms: Optional[torchvision.transforms.Compose] = None,
+                 validator: Optional[TimeSeriesForecastingInputValidator] = None,
+                 lagged_value: Optional[List[int]] = None,
+                 n_prediction_steps: int = 1,
+                 dataset_name: Optional[str] = None,
+                 normalize_y: bool = False,
                  ):
         """
+        tasks, the target_variables indicates which values in X corresponds to Y.
+        :param freq: Optional[Union[str, int]] frequency of the series sequences, used to determine the (possible)
+        period
```

Comment:
Could you adapt this docstring according to google's docstring style? you can find more information in the repo's wiki.